diff -u -r source-orig/src/lib/Hydra/Plugin/GithubPulls.pm source/src/lib/Hydra/Plugin/GithubPulls.pm
--- source-orig/src/lib/Hydra/Plugin/GithubPulls.pm     2018-03-11 19:13:49.514312424 -0300
+++ source/src/lib/Hydra/Plugin/GithubPulls.pm  2018-03-11 19:15:27.286672071 -0300
@@ -58,6 +58,7 @@
     system("jq -S . < $filename > $tempdir/github-pulls-sorted.json");
     my $storePath = `nix-store --add "$tempdir/github-pulls-sorted.json"`
         or die "cannot copy path $filename to the Nix store.\n";
+    chomp $storePath;
     my $timestamp = time;
     return { storePath => $storePath, revision => strftime "%Y%m%d%H%M%S", gmtime($timestamp) };
 }

