diff --git a/src/script/hydra-eval-jobset b/src/script/hydra-eval-jobset
index 9d375c1b..711a1df4 100755
--- a/src/script/hydra-eval-jobset
+++ b/src/script/hydra-eval-jobset
@@ -746,7 +746,7 @@ sub checkJobsetWrapped {
             # Wake up hydra-queue-runner.
             my $lowestId;
             while (my ($id, $x) = each %buildMap) {
-                $lowestId = $id if $x->{new} && (!defined $lowestId || $id < $lowestId);
+                $lowestId = $id if (!defined $lowestId || $id < $lowestId);
             }
             $notifyAdded->execute($lowestId) if defined $lowestId;
 
